feat: use rfcauthors for nomcom eligibility

jennifer-richards opened the pull request:
Refactors computation of the authorship path for nomcom eligibility to fix up some stale logic. Uses `RfcAuthor` records instead of `DocumentAuthor` for RFCs that have the former.

jennifer-richards committed:
refactor: author-based eligibility cleanup

jennifer-richards committed:
feat: use rfcauthor recs for nomcom eligbility

jennifer-richards committed:
chore: remove commented code

rjsparks reviewed (approved):
This will work best if the creation of rfcauthors via merged rfc and datatracker data preserves datatracker person links for authors on the last draft that remained authors on the RFC. I believe that merge has that property, but if it doesn't we should look to fix it.

jennifer-richards committed:
Merge branch 'feat/rfcauthors' into rfcauthor-eligibility

jennifer-richards committed:
refactor: factor out helper for testing

jennifer-richards committed:
test: test_get_qualified_author_queryset

jennifer-richards committed:
fix: restore a necessary import

jennifer-richards committed:
test: fix test_elig_by_author

jennifer-richards committed:
test: fix test_decorate_volunteers_with_qualifications

jennifer-richards committed:
test: add comment

jennifer-richards committed:
fix: drop test for draft-rfceditor state

Attempted to limit to drafts literally in the queue, but was not a
valid check when looking back in time. As a practical matter, the test
is not necessary.

jennifer-richards committed:
fix: exclude double counting, not rfc pub state

jennifer-richards committed:
test: update test
